modules/ravestate_roboyio/__init__.py

```diff
@@ -28,19 +28,30 @@
 
 if PYROBOY_AVAILABLE:
 
-    with Module(name="roboyio"):
+    ros2.set_node_once(node)
 
-        recognized_speech = Ros2SubProperty(
+    with rs.Module(name="roboyio"):
+
+        recognized_speech = ros2.Ros2SubProperty(
             name="recognized_speech",
             topic="/roboy/cognition/speech/recognition",
             msg_type=RecognizedSpeech,
             always_signal_changed=True)
 
-        @state(read=recognized_speech.id())
-        def roboy_input(ctx: ContextWrapper):
-            handle_single_interlocutor_input(ctx, ctx[recognized_speech.id()].text)
+        @rs.state(cond=recognized_speech.changed(), read=(interloc.prop_all, recognized_speech))
+        def roboy_input(ctx: rs.ContextWrapper):
+            result = ctx[recognized_speech]
+            if result:
+                interloc.handle_single_interlocutor_input(ctx, result.text)
+
+        # @rs.state(cond=startup(), read=interloc.prop_all)
+        # def roboy_input(ctx: ContextWrapper):
+        #     while not ctx.shutting_down():
+        #         result = listen()
+        #         if result:
+        #             interloc.handle_single_interlocutor_input(ctx, result)
 
-        @state(read="rawio:out")
+        @rs.state(read="rawio:out")
         def roboy_output(ctx):
-            ret = say(ctx["rawio:out:changed"])
-            logger.info(str(ret))
+            ret = say(unidecode(ctx["rawio:out:changed"]))
```

Comment:
Should be changed to rawio.prop_out.changed()

Reply:
Yes